Marlin/src/Marlin.cpp

```diff
@@ -1129,9 +1129,7 @@ void setup() {
  */
 void loop() {
 
-  for (;;) {
```

Comment:
The `for` loop is expected to perform better on AVR, if not on all platforms.

Reply:
@thinkyhead I know it is and I understand the reason (no jump in/out from procedure pushish/popping registers to stack at each loop) but that loop has two problems:
1) Messes with `debug_init_break  = tbreak loop` which I use in platformio.ini
2) Messes with platform logic loop.
the loop actually is itself called in a for (;;) but some platforms might do something else betwen those calls:

For example this is in STM32
```
int main(void)
{
  initVariant();
  setup();
  for (;;) {
#if defined(CORE_CALLBACK)
    CoreCallback();
#endif
    loop();
    if (serialEventRun) {
      serialEventRun();
    }
  }
  return 0;
}
```
An infinite loop inside loop, despite being a few clocks "faster", shouldn't be done 'cause it might introduce unwanted behavior in some platforms.

Reply:
I found this while developing the feature but I think I'll make a separate PR for this, since it's better to explain why it's necessary and keep a reference on it.